lib/vanagon/component/dsl.rb

```diff
@@ -333,6 +333,13 @@ def ref(the_ref)
         @component.options[:ref] = the_ref
       end
 
+      # Sets the clone options for git repository
+      #
+      # @param the_clone_options [Map] branch, depth to checkout for a git source
+      def clone_options(clone_options)
+        @component.options[:clone_options] = clone_options
+      end
```

Comment:
I think it would be more consistent with the rest of the vanagon codebase if we had clone_option take in an option and the value we want that set to, and use that to populate the `:clone_options` hash. Maybe something like this (I haven't had a chance to test yet)
```suggestion
      def clone_option(option, value)
        option = option.to_sym
        @component.options[:clone_options][option] = value
      end
```

Then in the component have something like
```
pkg.clone_option 'branch', 'foo'
pkg.clone_option 'depth', 50
```

If you have the ability to test out this suggestion that would be great, if not I can take a look in the next day or two

Reply:
It works and test added